DOCS-8072-4.3-tuning-profiles-kt (#1424)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/tuning-pooling-profiles.adoc

> = Pooling Profiles

Reviewer: ```suggestion
= Request Management Using Pooling Profiles
```

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Unlike singleton components, pooled components configure their component pool which contains multiple instances of the component to handle simultaneous incoming requests. Each pooled component has its pooling profile.

Reviewer: This is confusing: "pooled components configure their component pool." 

Do we actually mean that configuring component pooling results in one group, or pool, of multiple instances of the same component (or instances of many different components?) so that the pool can process multiple incoming requests simultaneously, instead of a single component processing one request at a time?

Or do we mean "connections" instead of "components," and that pooling connections enables processing multiple incoming requests simultaneously?

Reviewer: This is also confusing: "Each pooled component has its pooling profile."

Does this really mean that each pool of components has a pooling profile,
or that each component in the pool has its own individual profile,
or that each component in the pool has the same profile as the rest of the components in that same pool?

Or does this mean that each pool of connections has a profile
or each connection in the pool has its own,
or each connection in the pool as the same as the rest?

I think the use of "component" in what is confusing. What is being pooled?

Reviewer: ```suggestion
Instead of using a single connection, improve request processing by Anypoint Connectors by pooling groups of individual connections to process simultaneous incoming requests. Each pool of connections has a unique pooling profile that defines the pool constituency.
```
Can we describe what this profile consists of and why it is necessary?

> == Use Pooling Profile Parameter

Reviewer: omit; unnecessary
```suggestion
```

> Connectors such as Anypoint Connector for FTP (FTP Connector) or Anypoint Connector for SFTP (SFTP Connector) pool the connections by default and use a pooling profile to customize the connection pool. Add the `pooling-profile` parameter to the connectors when performance tests show it necessary.

Reviewer: ```suggestion
Connectors such as Anypoint Connector for FTP (FTP Connector) and Anypoint Connector for SFTP (SFTP Connector) pool connections by default and use a pooling profile to describe the connection pool. 
```
But this is confusing: "Add the `pooling-profile` parameter to the connectors when performance tests show it necessary."
If these connectors pool connections by default, WHY would a user need to add the profile?
And are we saying that adding this parameter actually creates the pool?

> However, Anypoint Connector for Database (Database Connector) does not perform connection pooling by default, which means that every execution of an SQL statement opens a new connection and closes it once finished (unless there is a transaction active that uses the same connection for the whole transaction). In this case, by using a pooling profile, the connection set-up occurs only once and subsequent requests use existing connections achieving a performance benefit as a result. See the following example:

Reviewer: ```suggestion
However, some connectors, such as Anypoint Connector for Database (Database Connector), do not automatically pool connections, causing every execution of an SQL statement to open a new connection to process requests and then close it once finished. By configuring a pooling profile, you enable one-time connection setup and connection reuse, thereby improving request-processing performance:
```

[source,xml,linenums]
----
<db:mysql-config name="dbConfig" host="localhost" port="3306" user="root"
   password="" database="esb" doc:name="MySQL Configuration">
   <db:pooling-profile maxPoolSize="17" minPoolSize="13" />
</db:mysql-config>
----

> The `maxPoolSize` setting specifies the maximum number of connections that will be open to the database, and that will remain available to serve incoming requests.

Reviewer: ```suggestion
The `maxPoolSize` setting specifies the maximum number of connections that remain available in the pool to serve incoming requests.
```
